mars/services/storage/core.py

```diff
@@ -353,38 +391,50 @@ async def _get_meta_api(self, session_id: str):
     def get_client_params(self):
         return self._init_params
 
-    def allocate_size(self,
-                      size: int,
-                      level: StorageLevel):
-        if self._quotas[level].request(size):
+    async def allocate_size(self,
+                            size: int,
+                            level: StorageLevel):
+        if await self._quotas[level].request(size):
             return
         else:  # pragma: no cover
-            raise NotImplementedError
+            raise NotImplementedError('Spill is not supported now')
+
+    async def fetch(self,
+                    session_id: str,
+                    data_key: str,
+                    level: StorageLevel,
+                    address: str,
+                    band: str):
+        from .transfer import SenderManagerActor
 
-    async def prefetch(self,
-                       session_id: str,
-                       data_key: str,
-                       level: StorageLevel):
         try:
             info = self._data_manager.get_info(session_id, data_key)
             self.pin(info.object_id)
-        except DataNotExist:  # pragma: no cover
-            # Not exist in local, fetch from remote worker
+        except DataNotExist:
+            # Not exists in local, fetch from remote worker
             try:
-                yield self._storage_handler.prefetch(session_id, data_key)
+                yield self._storage_handler.fetch(session_id, data_key)
             except NotImplementedError:  # pragma: no cover
-                raise
-
-    def update_quota(self,
-                     size: int,
-                     level: StorageLevel):
-        self._quotas[level].update(size)
-
-    def release_size(self,
-                     size: int,
-                     level: StorageLevel
-                     ):
-        self._quotas[level].release(size)
+                meta_api = await self._get_meta_api(session_id)
```

Comment:
I think we need some comments here or change the exception raised, as other contributors may think we may implement `fetch` under this scenario someday.